torch/masked/maskedtensor/passthrough.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates
+"""
+These are functions that should simply be applied to both mask and data.
+Take select or stack as an example. This operation can be applied to
+both the mask and data of a MaskedTensor and the result wrapped into
+a new MaskedTensor as a result.
+"""
+
+import torch
+
+__all__ = []  # type: ignore[var-annotated]
+
+
+PASSTHROUGH_FNS = [
+    torch.ops.aten.select,
+    torch.ops.aten.transpose,
+    torch.ops.aten.split,
+    torch.ops.aten.t,
+    torch.ops.aten.slice,
+    torch.ops.aten.slice_backward,
+    torch.ops.aten.select_backward,
+    torch.ops.aten.index,
+    torch.ops.aten.expand,
+    torch.ops.aten.view,
+    torch.ops.aten._unsafe_view,
+    torch.ops.aten._reshape_alias,
+    torch.ops.aten.cat,
+]
+
+
+# TODO: tree_map doesn't cut it due to kwargs support for torch_function
+def _map_mt_args_kwargs(args, kwargs, map_fn):
+    if kwargs is None:
+        kwargs = {}
+    impl_args = []
+    for a in args:
+        from .core import is_masked_tensor
+
+        if is_masked_tensor(a):
+            impl_args.append(map_fn(a))
+        elif torch.is_tensor(a):
+            impl_args.append(a)
+        elif isinstance(a, list):
+            a_impl, _ = _map_mt_args_kwargs(a, {}, map_fn)
+            impl_args.append(a_impl)
+        elif isinstance(a, tuple):
+            a_impl, _ = _map_mt_args_kwargs(a, {}, map_fn)
+            impl_args.append(tuple(a_impl))
+        else:
+            impl_args.append(a)
+    impl_kwargs = {
+        k: map_fn(v) if is_masked_tensor(v) else v for (k, v) in kwargs.items()
```

Comment:
above we're recursing into list and tuples for args, presumably we'd want to do the same for kwargs, no?

Reply:
Yeah that makes sense, fixed